pkg/plugin/driver/daemonset/daemonset.go

```diff
@@ -78,19 +85,46 @@ func (p *Plugin) GetResultType() string {
 	return p.Definition.ResultType
 }
 
+//FillTemplate populates the internal Job YAML template with the values for this particular job.
+func (p *Plugin) FillTemplate(hostname string) (*bytes.Buffer, error) {
```

Comment:
I'd expect the return type to be `([]byte, error)`, then line 94 & 107 could return `nil, wrapped error`.

Might be good to add a comment explaining why we use `*bytes.Buffer`.

Also applies to job driver.